excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java

```diff
@@ -60,14 +60,39 @@ public static void register(ReaderRegistry registry) {
 
   @Override
   public Table read(XlsxReadOptions options) throws IOException {
-    List<Table> tables = readMultiple(options);
-    if (tables.isEmpty()) {
-      throw new IllegalArgumentException("No tables found.");
+    List<Table> tables = readMultiple(options, true);
+    if (options.sheetIndex() != null) {
+      int index = options.sheetIndex();
+      if (index < 0 || index >= tables.size()) {
+        throw new IndexOutOfBoundsException(
+            "Sheet index outside bounds. " + tables.size() + " sheets found.");
```

Comment:
sorry. one last thing I just noticed. How about also adding the index here:

    "Sheet index " + index + " outside bounds. " + tables.size() + " sheets found.");

Reply:
Done. If you want I can also add a test for this:

`assertEquals("Sheet index 5 outside bounds. 2 sheets found.", iobe.getMessage());`